magma/t.py

```diff
@@ -18,6 +18,7 @@ def __init__(self, name=None):
             name = AnonRef()
         elif isinstance(name, str):
             name = NamedRef(name=name)
+            name.value = self
```

Comment:
Should just use the constructor interface here?

Reply:
Good catch. Also using a weakref to avoid circular reference